### PR TITLE
Platform 935 analytics upgrade phase2

### PR DESCRIPTION
**Pull request by harnash**
Phase 2 of migration to Universal Analytics this includes:
- allow enabling Universal Analytics on a specific wiki through Wiki Factory's variable
- cleanup of Tracker::track() arguments (method parameter now accepts only 'analytics', 'internal' and 'ad' values)
- initial custom dimensions layout for Universal Analytics
- all data is sent to one Google Analytics account with UA enabled (for testing purposes)
- legacy Google Analytics is still enabled for all wiki's
- custom dimensions for AB Tests will not work since non-Premium Google Analytics Accounts can't define more than 20 custom dimensions

Config PR: https://github.com/Wikia/config/pull/1023

### CODE REVIEW

File: extensions/wikia/AnalyticsEngine/AnalyticsEngine.php [hunk 1]
```diff
@@ -43,6 +43,8 @@ private static function getProvider($provider) {
 				return new AnalyticsProviderExelate();
 			case 'GAS':
 				return new AnalyticsProviderGAS();
+            case 'GoogleUA':
+                return new AnalyticsProviderGoogleUA();
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsEngine.php]: Mixed spaces and tabs

File: extensions/wikia/AnalyticsEngine/AnalyticsEngine.setup.php [hunk 2]
```diff
@@ -25,18 +25,23 @@
...
 $wgHooks['InstantGlobalsGetVariables'][] = 'AnalyticsProviderIVW2::onInstantGlobalsGetVariables';
 
 //register hook for WikiaMobile skin to get the asset as part of the head js package in one request
 $wgHooks['WikiaMobileAssetsPackages'][] = 'AnalyticsProviderGAS::onWikiaMobileAssetsPackages';
+$wgHooks['WikiaMobileAssetsPackages'][] = 'AnalyticsProviderGoogleUA::onWikiaMobileAssetsPackages';
 $wgHooks['WikiaMobileAssetsPackages'][] = 'AnalyticsProviderBlueKai::onWikiaMobileAssetsPackages';
 
 // register hooks for Venus
 $wgHooks['VenusAssetsPackages'][] = 'AnalyticsProviderGAS::onVenusAssetsPackages';
+$wgHooks['VenusAssetsPackages'][] = 'AnalyticsProviderGoogleUA::onVenusAssetsPackages';
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsEngine.setup.php]: Venus skin is no longer supported. No need to add a hook for it.

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGoogleUA.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		return true;
+	}
+
+	static public function onVenusAssetsPackages( &$jsHeadGroups, &$jsBodyGroups, &$cssGroups) {
+		if (self::isEnabled()) {
+			$jsHeadGroups[] = 'universal_analytics_js';
+		}
+
+		return true;
+	}
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGoogleUA.php]: `onVenusAssetsPackages` can be removed

File: extensions/wikia/AnalyticsEngine/js/universal_analytics.js [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+        );
+    } else {
+        window.ga(
+            'create', 'UA-32129070-3', 'auto',
+            {
+                'name': 'ads',
+                'sampleRate': 100,
+                'allowLinker': true,
+                'userId': window.wgGAUserIdHash
+            }
```
Comment (macbre) [inline extensions/wikia/AnalyticsEngine/js/universal_analytics.js]: The same account ID for both dev and production?

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderGoogleUA.php (hunk 3 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderGoogleUA.php]: Fixed in ab756c4d4132b215f2d10bcdb6133839a9b15adc

File: extensions/wikia/AnalyticsEngine/AnalyticsEngine.setup.php (hunk 2 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/AnalyticsEngine.setup.php]: Fixed in ab756c4d4132b215f2d10bcdb6133839a9b15adc

File: extensions/wikia/AnalyticsEngine/js/universal_analytics.js (hunk 4 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/js/universal_analytics.js]: Right. Added different accounts for prod and dev 6a371d630e2bf012b121e2d0a74b38c29c94f158

File: extensions/wikia/AnalyticsEngine/AnalyticsEngine.php (hunk 1 above)
Comment (harnash) [inline extensions/wikia/AnalyticsEngine/AnalyticsEngine.php]: Fixed in 9f337a4be671c4cddd7962889a4fa6378e211093

**Comment by macbre** [comment]
LGTM

**Pull request merged**